### PR TITLE
Avoid having small right operand on its own line

### PR DESCRIPTION
**Pull request by jablko**
Would you consider grouping the right operand, like [language-js](https://github.com/prettier/prettier/blob/master/src/language-js/printer-estree.js#L5899), "to avoid having a small right part like -1 be on its own line"?

### CODE REVIEW

**Comment by Janther** [comment]
prettier js formats the following:
```Javascript
veryVeryVeryVeryVeryLongVariableCalledA - veryVeryVeryVeryVeryLongVariableCalledB - 1 -1 -1 - 1 - 1 - 1 - 1;
```
like,
```Javascript
veryVeryVeryVeryVeryLongVariableCalledA -
  veryVeryVeryVeryVeryLongVariableCalledB -
  1 -
  1 -
  1 -
  1 -
  1 -
  1 -
  1;
```
have a look at #175 to see other ways prettier js groups ecuations that we are reviewing right now for solidity.

**Comment by jablko** [comment]
My mistake, however it formats
```JavaScript
a(veryVeryVeryVeryVeryLongVariableCalledA + veryVeryVeryVeryVeryLongVariableCalledB) - 1;
```
like
```JavaScript
a(
  veryVeryVeryVeryVeryLongVariableCalledA +
    veryVeryVeryVeryVeryLongVariableCalledB
) - 1;
```

Would you consider something like the revised version of this pull request, which handles both
```Solidity
veryVeryVeryVeryVeryLongVariableCalledA -
  veryVeryVeryVeryVeryLongVariableCalledB -
  1 -
  1 -
  1 -
  1 -
  1 -
  1 -
  1;
```
and
```Solidity
a(
  veryVeryVeryVeryVeryLongVariableCalledA +
    veryVeryVeryVeryVeryLongVariableCalledB
) - 1;
```
?

**Comment by Janther** [comment]
hahaha this one is a weird one.
If I try this in JS:
```Javascript
2 - a(veryVeryVeryVeryVeryLongVariableCalledA + veryVeryVeryVeryVeryLongVariableCalledB) - 1;

a(veryVeryVeryVeryVeryLongVariableCalledA + veryVeryVeryVeryVeryLongVariableCalledB) - 1 + 2;
```
I get
```Javascript
2 -
  a(
    veryVeryVeryVeryVeryLongVariableCalledA +
      veryVeryVeryVeryVeryLongVariableCalledB
  ) -
  1;

a(
  veryVeryVeryVeryVeryLongVariableCalledA +
    veryVeryVeryVeryVeryLongVariableCalledB
) -
  1 +
  2;
```

I like having this particular edge case as long as the main rule is not broken, could you add test cases??
testing not only your solution but also combining it in different scenarios to check exactly how it behaves not just in a specific case but also in something broader.

File: src/binary-operator-printers/comparison.js [hunk 1]
```diff
@@ -23,14 +23,16 @@ module.exports = {
     const groupIfNecessary = groupIfNecessaryBuilder(path);
     const indentIfNecessary = indentIfNecessaryBuilder(path);
 
+    const right = concat([node.operator, line, path.call(print, 'right')]);
+    const parent = path.getParentNode();
+    const shouldGroup =
+      parent.type !== node.type && node.left.type !== node.type;
```
Comment (Janther) [inline src/binary-operator-printers/comparison.js]: Could you also document why this particular type mismatch deserves a grouping?

Comment (jablko) [reply src/binary-operator-printers/comparison.js]: :+1: Will do!

**Comment by jablko** [comment]
I've rebased on #182: I added tests for
- single operator/two operands
- multiple operators
- arithmetic, comparison, and logical operators
- no context, assignment, if statement, function call, and return statement

**Comment by mattiaerre** [comment]
how can we fix this cov report issue? I'll try to reach out to the Codecov team and see what they say

**Comment by mattiaerre** [comment]
can you resolve the conflicts here too @jablko ? thanks

**Comment by jablko** [comment]
@mattiaerre Absolutely: Rebased.

**Comment by mattiaerre** [comment]
thanks os much @jablko

File: tests/BinaryOperators/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -330,8 +326,7 @@ contract ArithmeticOperators {
                 veryVeryVeryVeryVeryLongVariableCalledB;
         veryVeryVeryVeryVeryLongFunctionCalledA(
             veryVeryVeryVeryVeryLongVariableCalledB
-        ) **
-            c;
+        )**c;
```
Comment (mattiaerre) [inline tests/BinaryOperators/__snapshots__/jsfmt.spec.js.snap]: just noticing that there is no space before `**` and after it; is that what we want?

Comment (Janther) [reply tests/BinaryOperators/__snapshots__/jsfmt.spec.js.snap]: this was applied in #174

Comment (mattiaerre) [reply tests/BinaryOperators/__snapshots__/jsfmt.spec.js.snap]: can we approve this PR?
